homeassistant/components/lcn/__init__.py

```diff
@@ -44,62 +35,89 @@
     VarReset,
 )
 
+PLATFORMS = ["binary_sensor", "climate", "cover", "light", "scene", "sensor", "switch"]
+
 _LOGGER = logging.getLogger(__name__)
 
 
-async def async_setup(hass, config):
-    """Set up the LCN component."""
-    hass.data[DATA_LCN] = {}
-
-    conf_connections = config[DOMAIN][CONF_CONNECTIONS]
-    connections = []
-    for conf_connection in conf_connections:
-        connection_name = conf_connection.get(CONF_NAME)
-
-        settings = {
-            "SK_NUM_TRIES": conf_connection[CONF_SK_NUM_TRIES],
-            "DIM_MODE": pypck.lcn_defs.OutputPortDimMode[
-                conf_connection[CONF_DIM_MODE]
-            ],
-        }
+async def async_setup_entry(hass, config_entry):
+    """Set up a connection to PCHK host from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
 
-        connection = pypck.connection.PchkConnectionManager(
-            conf_connection[CONF_HOST],
-            conf_connection[CONF_PORT],
-            conf_connection[CONF_USERNAME],
-            conf_connection[CONF_PASSWORD],
+    settings = {
+        "SK_NUM_TRIES": config_entry.data[CONF_SK_NUM_TRIES],
+        "DIM_MODE": pypck.lcn_defs.OutputPortDimMode[config_entry.data[CONF_DIM_MODE]],
+    }
+
+    # connect to PCHK
+    if config_entry.entry_id not in hass.data[DOMAIN]:
+        lcn_connection = pypck.connection.PchkConnectionManager(
+            config_entry.data[CONF_IP_ADDRESS],
+            config_entry.data[CONF_PORT],
+            config_entry.data[CONF_USERNAME],
+            config_entry.data[CONF_PASSWORD],
             settings=settings,
-            connection_id=connection_name,
+            connection_id=config_entry.title,
```

Comment:
I wouldn't use the title for an api parameter. The title should just be user facing.